src/core/tests/Test_StatusHandler.py

```diff
@@ -520,6 +532,42 @@ def test_load_status_and_set_package_install_status(self):
         self.assertTrue('Critical' in str(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"][2]["classifications"]))
         self.runtime.env_layer.file_system.delete_files_from_dir(self.runtime.status_handler.status_file_path, '*.complete.status')
 
+    def test_remove_old_complete_status_files(self):
+        """ Create dummy files in status folder and check if the complete_status_file_path is the latest file and delete those dummy files """
+        file_path = self.runtime.execution_config.status_folder
+        for i in range(1, 15):
+            with open(os.path.join(file_path, str(i + 100) + '.complete.status'), 'w') as f:
+                f.write("test" + str(i))
+
+        packages, package_versions = self.runtime.package_manager.get_all_updates()
+        self.runtime.status_handler.set_package_assessment_status(packages, package_versions)
+        self.runtime.status_handler.load_status_file_components(initial_load=True)
+
+        # remove 10 complete status files
+        count_status_files = glob.glob(os.path.join(file_path, '*.complete.status'))
+        self.assertEqual(10, len(count_status_files))
+        self.assertTrue(os.path.isfile(self.runtime.execution_config.complete_status_file_path))
+        self.runtime.env_layer.file_system.delete_files_from_dir(file_path, '*.complete.status')
+        self.assertFalse(os.path.isfile(os.path.join(file_path, '1.complete_status')))
+
+    def test_remove_older_complete_status_files(self):
```

Comment:
rename to test_remove_old_complete_status_files_throws_exception()

Reply:
done